library/src/main/java/com/novoda/downloadmanager/DownloadError.java

```diff
@@ -11,13 +12,24 @@
         UNKNOWN
     }
 
-    private final Error error;
+    private final Type type;
+    private final String message;
 
-    DownloadError(Error error) {
-        this.error = error;
+    DownloadError(Type type, String message) {
+        this.type = type;
+        this.message = message;
     }
 
-    Error error() {
-        return error;
+    DownloadError(Type type) {
+        this.type = type;
+        this.message = "";
+    }
+
+    Type error() {
+        return type;
+    }
+
+    public String message() {
```

Comment:
This one and `error` probably should be `getXXX` to make it a Kotlin property when used with it. Also, should the getter be `type` instead of error? We may need to leave the current methods for convenience or compatibility (Maybe deprecated?)

Reply:
Yea this should be `type` 👍 I'm not concerned about convenience of compatibility atm, version two hasn't been released. The app we are using this in I need to supply this extra information in anyway, so will be easily be able to update.